packages/riverpod_lint/lib/src/lints/avoid_use_ref_inside_dispose.dart

```diff
@@ -0,0 +1,51 @@
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/error/listener.dart';
+import 'package:collection/collection.dart';
+import 'package:custom_lint_builder/custom_lint_builder.dart';
+import 'package:riverpod_analyzer_utils/riverpod_analyzer_utils.dart';
+
+import '../riverpod_custom_lint.dart';
+
+const overrideAnnotation = 'override';
+const disposeMethod = 'dispose';
+
+class AvoidUseRefInsideDispose extends RiverpodLintRule {
+  const AvoidUseRefInsideDispose() : super(code: _code);
+
+  static const _code = LintCode(
+    name: 'avoid_use_ref_inside_dispose',
+    problemMessage: "Avoid using 'Ref' in the dispose method.",
```

Comment:
I think the following name would be more logical

```suggestion
    name: 'avoid_ref_inside_state_dispose',
```